### ISSUE TITLE
Unable to access Azure DevOps - Unable to clear cached credentials

### ISSUE THREAD

**Issue by IIStevowII** [GitCredentialManager/git-credential-manager#724]
**Which version of GCM are you using?**
2.0.696+4365b917da

**Which Git host provider are you trying to connect to?**

* [X] Azure DevOps
* [ ] Azure DevOps Server (TFS/on-prem)
* [ ] GitHub
* [ ] GitHub Enterprise
* [ ] Bitbucket
* [ ] Other - please describe

**Can you access the remote repository directly in the browser using the remote URL?**

* [X] Yes (Also when using Private Mode in the Browser)
* [ ] No, I get a permission error
* [ ] No, for a different reason - please describe

---

**_[Azure DevOps only]_ What format is your remote URL?**

* [ ] Not applicable
* [ ] https://dev.azure.com/`{org}`/...
* [X] https://`{org}`@dev.azure.com/`{org}`/...
* [ ] https://`{org}`.visualstudio.com/...

**_[Azure DevOps only]_ If the account picker shows more than one identity as you authenticate, check that you selected the same one that has access on the web.**

* [X] Not applicable
* [ ] I only see one identity
* [ ] I checked each identity and none worked

---

**Expected behavior**

I should receive a browser popup to authenticate, or the command line should print out a device code for me to authenticate. After this I should be succesfully authenticate.

**Actual behavior**

When trying to make a connection to Azure DevOps with GIT, i am getting authentication/SSL errors. Using Visual Studio to clone the Azure Repo, I sometimes have some success with cloning the repository. But a pull after that results in the same errors. Using the CMD always results in connection errors.

My current .gitconfig file:
```
http.sslbackend=schannel
credential.azreposcredentialtype=oauth
credential.usehttppath=true
credential.helper=manager-core
credential.https://dev.azure.com.provider=azure-repos
credential.https://dev.azure.com.usehttppath=true
```

Changed the git config setting `http.sslbackend` to `openssl` gives the following error:
`fatal: unable to access 'https://dev.azure.com/stevenhack/Home%20Assistant/_git/Home.Assistant/': OpenSSL SSL_connect: Connection was reset in connection to dev.azure.com:443`

What interests me a lot is that I do not get a browser popup to authenticate, or a device code or whatever... For some reason it seems to already have credentials, but I've tried several ways to clear anything:

1) Looked a credential manager -> no GIT related credentials stored that
2) Ran the command: `git credential-manager-core azure-repos list [organization]` -> outputs a single blank line as output.
3) Ran the command: `git credential-manager-core azure-repos list --show-remotes` -> outputs the following:
[organization]:
  origin:
    (fetch) -> (inherit)
    (push)  -> (inherit)

I have also tried to (with reboots in between ofcourse)
1) Uninstall + install GCM -> no luck
2) Uninstalled + Installed GIT -> no luck

Running Windows 11, Git + GCM installed + Visual Studio 2019 and 2022 + Visual Studio Code
I also ran the AzureDevOpsTls12Analysis.ps1 script from https://github.com/microsoft/azure-devops-tls12 but everything seems to be OK.

**Logs**

```
C:\Workspace\[project]\[repository]>git pull
22:29:02.446843 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
22:29:02.446843 git.c:459               trace: built-in: git pull
22:29:02.446843 run-command.c:654       trace: run_command: git fetch --update-head-ok
22:29:02.452866 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
22:29:02.462891 git.c:459               trace: built-in: git fetch --update-head-ok
22:29:02.462891 run-command.c:654       trace: run_command: GIT_DIR=.git git remote-https origin 'https://[organization]@dev.azure.com/[organization]/[project]/_git/[repository]'
22:29:02.462891 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
22:29:02.478564 git.c:748               trace: exec: git-remote-https origin 'https://[organization]@dev.azure.com/[organization]/[project]/_git/[repository]'
22:29:02.478564 run-command.c:654       trace: run_command: git-remote-https origin 'https://stevenhack@dev.azure.com/stevenhack/Home%20Assistant/_git/Home.Assistant'
22:29:02.478564 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
fatal: unable to access 'https://dev.azure.com/[organization]/[project]/_git/[repository]/': OpenSSL SSL_connect: Connection was reset in connection to dev.azure.com:443
```

**Comment by ldennington** [comment]
The logs don't indicate Git is calling GCM. This may be a Git for Windows issue instead - @dscho, do you have any thoughts? Also can you include the output from the AzureDevOpsTls12Analysis.ps1 script?

**Comment by dscho** [comment]
> OpenSSL SSL_connect: Connection was reset in connection to dev.azure.com:443

This looks like a regular network problem to me. In the past, I saw similar messages e.g. when a network adapter or access point was faulty, or when an unreliable proxy was in use. I can't say that this is the case here, but if OpenSSL has trouble because a connection was reset, there is very little Git or GCM can do, the fault usually lies elsewhere.

**Comment by ldennington** [comment]
Closing due to no response/not a GCM issue.

**Comment by IIStevowII** [comment]
Sorry for the late replies, I have tried several things to rule out. Eventually I came across several other issues that seem to be the same:
https://developercommunity.visualstudio.com/t/connection-problems-with-azure-devops-for-at-least/1433222
https://developercommunity.visualstudio.com/t/err-connection-reset-when-trying-to-access-devazur/1204094

The fix suggested in these topics is a following: `netsh interface ipv6 set prefixpolicy ::ffff:0:0/96 46 4`

This seemed to do the trick, GCM now comes with the desired popup for authentication. So you guys were right, it seemed to be a connection issue regarding IPV6.

**Comment by dscho** [comment]
> So you guys were right

That's incorrect. Half of those who were right aren't guys. Let's give credit where credit is due!

**Comment by IIStevowII** [comment]
English is not my native language, I didn't mean to imply a gender... i was going for a "You all" or something like that. (which in turn sounds Texan)
